features/step_definitions/basic_steps.rb

```diff
@@ -10,21 +14,34 @@
   end
 end
 
+When(/^I fill in:$/) do |table|
+  table.hashes.each do |hash|
+      fill_in hash[:element], with: hash[:content]
+  end
+end
+
+When(/^I click the "([^"]*)" button$/) do |button|
+  click_button button
+end
+
+Then(/^I should see "([^"]*)"$/) do |content|
+  expect(page).to have_content content
+end
+
 When(/^I click the link "([^"]*)"$/) do |link|
   click_link(link)
 end
 
-Then(/^I should be on the "([^"]*)" page$/) do |page|
-  expected_page = goto(page)
-  expect(current_path).to eq expected_page
```

Comment:
They were not the same. One navigated to a page the other asserted it was on a page.  Of course all the test go green.  If you navigate to a page and asset that you are on the page in the same definition the test will go green automatically.  I asked you not to have assertions in navigatio steps